### DISCUSSION TITLE
Cannot use generic values in defineStore with setup stores

### DISCUSSION THREAD

**Discussion by Q16solver** [vuejs/pinia#2069] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/github-wn9aqn-1ldb7t

### Steps to reproduce the bug

1. Run `npm i`
2. Go to `store/error.ts` to see the typescript error

### Expected behavior

Should have no typescript errors and be able to useStore normally after specifying generic with setup stores

### Actual behavior

Getting typescript error, maybe it is unable to unwrap types because of the generic

**Comment by posva** [comment, marked as answer]
This is indeed a limitation Vue refs due to unwrapping, note it has been mentioned in a couple issues before. Moving to a discussion

**Reply by Q16solver** [reply]
Ah Ok, I think I saw some other previous similar issues, but they were for old syntax pinia stores. It seems like the solution for that was to use old syntax and simply cast as Ref as noted in https://github.com/vuejs/pinia/discussions/973 and it appears to work, is that the only workaround we have? If so, I'll probably rewrite it to use the old syntax instead of setup stores, do you also mind linking me to any issue/docs that contain information about this limitation? Would be great reference and learning for me, thanks! 😊

**Reply by posva** [reply]
You can find the issues [here](https://github.com/vuejs/core/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+unwrap) 🙂
